dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance/test/AsyncTest.java

```diff
@@ -204,6 +202,46 @@ public void testAsyncRequestContextDeactivatedOnInterruption() throws Interrupte
         }
     }
 
+    @Test
+    public void testAsyncRequestContextDeactivatedOnInterruption() throws InterruptedException, ExecutionException, TimeoutException, TestException {
+        ExecutorBuilder<String> builder = FaultToleranceProvider.newExecutionBuilder();
+
+        MockAsyncRequestContextController asyncRequestContextController = new MockAsyncRequestContextController();
+        builder.setRequestContextController(asyncRequestContextController);
+
+        Executor<Future<String>> executor = builder.buildAsync(Future.class);
+        ExecutionContext context = executor.newExecutionContext("testAsyncCS", null);
+
+        CountDownLatch startSignal = new CountDownLatch(1); // Prevents the execution from being interrupted until request context is set
+
+        // Test an execution which is cancelled
+        Future<String> result = executor.execute(() -> {
+            assertEquals("Activate request context should be called when request is started", 1, asyncRequestContextController.getActivateContextCount());
+
+            // Allow the execution to be interrupted
+            startSignal.countDown();
+
+            // Wait for up to 10 seconds for the execution to be interrupted
+            Thread.sleep(FUTURE_TIMEOUT);
+
+            return null;
+        }, context);
+
+        // Wait for the execution to have started before interrupting
+        startSignal.await();
```

Comment:
Add a timeout here as we don't want to wait forever. Also then assert that it returned `true`.